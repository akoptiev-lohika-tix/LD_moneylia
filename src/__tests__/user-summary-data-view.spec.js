import React from 'react';
import { render, screen, waitFor } from '@testing-library/react-native';
import renderer from 'react-test-renderer';

import UserSummaryDataView from '../components/user-summary-data-view/user-summary-data-view';

const props = {
  title: 'Test title',
  data: 2022,
};

test('UserSummaryDataViewTesting renders correctly', async () => {
  const tree = await waitFor(() =>
    renderer.create(<UserSummaryDataView title={props.title} data={props.data} />).toJSON()
  );
  expect(tree).toMatchSnapshot();
});

describe('UserSummaryDataViewTesting', () => {
  beforeEach(async () => {
    await waitFor(() => render(<UserSummaryDataView title={props.title} data={props.data} />));
  });

  it('should have correct title', () => {
    expect(screen.findByText(props.title)).toBeTruthy();
  });

  it('should have correct data', () => {
    expect(screen.findByText(props.data)).toBeTruthy();
  });
});
