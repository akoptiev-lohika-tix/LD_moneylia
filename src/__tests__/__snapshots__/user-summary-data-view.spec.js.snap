// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSummaryDataViewTesting renders correctly 1`] = `
<View
  style={
    Object {
      "marginRight": 0,
      "width": 140,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#6f6d7b",
        "fontFamily": "Work Sans Regular",
        "fontSize": 12,
        "lineHeight": 14,
        "marginBottom": 4,
      }
    }
  >
    Test title
  </Text>
  <Text
    style={
      Object {
        "color": "#312e43",
        "fontFamily": "Work Sans Regular",
        "fontSize": 16,
        "lineHeight": 19,
      }
    }
  >
    2022
  </Text>
</View>
`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "marginRight": 0,
      "width": 140,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#6f6d7b",
        "fontFamily": "Work Sans Regular",
        "fontSize": 12,
        "lineHeight": 14,
        "marginBottom": 4,
      }
    }
    testID="data-view-title"
  >
    Test title
  </Text>
  <Text
    style={
      Object {
        "color": "#312e43",
        "fontFamily": "Work Sans Regular",
        "fontSize": 16,
        "lineHeight": 19,
      }
    }
  >
    2022
  </Text>
</View>
`;
